### PR TITLE
Refactor `verify_aptrust` to use more efficient query (DBAI-53)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR aims to resolve DBAI-53.

### CODE REVIEW

File: test/test_status_event_repository.rb [hunk 1]
```diff
@@ -100,56 +100,81 @@ def test_get_latest_event_for_bag_when_nil
...
+  end
+
+  def test_get_latest_event_for_bags_with_no_start_time
+    create_test_data
+
+    events = mixin_repo.get_latest_event_for_bags
+
+    assert_equal 4, events.length
+    unique_bag_identifiers = events.map { |e| e.bag_identifier }.uniq
+    assert_equal 4, unique_bag_identifiers.length
```
Comment (ssciolla) [inline test/test_status_event_repository.rb:177]: Do I need to check anything else here? Not sure.

File: test/test_status_event_repository.rb [hunk 2]
```diff
@@ -100,56 +100,86 @@ def test_get_latest_event_for_bag_when_nil
...
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_two, timestamp: start_time + 180)
-    mixin_repo.create(status: BagStatus::FAILED, bag_identifier: bag_identifier_two, timestamp: start_time + 200)
+  def test_get_latest_event_for_bags_when_no_events
+    start = Time.utc(2024, 5, 4, 12, 0, 0, 0)
+    bag_events = mixin_repo.get_latest_event_for_bags(start_time: start)
+    assert_equal 0, bag_events.length
+  end
 
+  def test_get_latest_event_for_bags
+    bag_id_one, bag_id_two, bag_id_three, bag_id_four = create_latest_event_for_bags_test_data
```
Comment (jayamala17) [inline test/test_status_event_repository.rb:146]: bag_id_one can be removed since its not used

Comment (ssciolla) [reply test/test_status_event_repository.rb:146]: I'm not sure how to do this when you're destructuring. Let me check.

Comment (ssciolla) [reply test/test_status_event_repository.rb]: Fixed. See https://github.com/mlibrary/bag-courier/pull/53/commits/56c048da06d398d13aba9e075e783fc68de3eaee

Comment (jayamala17) [reply test/test_status_event_repository.rb]: if i remove bag_id_one its failing the test but i get that variable unused error. can we check the assert for bag_id_one or we can ignore this error?

**Review by jayamala17 — APPROVED** [review]

File: test/test_status_event_repository.rb (hunk 2 above)
Comment (ssciolla) [reply test/test_status_event_repository.rb]: I'm pretty sure I fixed it in my last commit. GitHub says tests are passing. Are you using the latest code? Or are you changing the wrong test? Cause the second one uses all four variables.

Comment (jayamala17) [reply test/test_status_event_repository.rb]: works with the latest

**Pull request merged**